dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04200CreateDefaultVanityURL.java

```diff
@@ -0,0 +1,235 @@
+
+	@Override
+	public void executeUpgrade() throws DotDataException, DotRuntimeException {
+		DotConnect dc = new DotConnect();
+
+		
+		//Inserts into Inode Table the reference of the Vanity URL Content Type 
+		dc.setSQL(INSERT_INODE_QUERY);
+		dc.addParam(DEFAULT_VANITY_URL_STRUCTURE_INODE);
+		dc.addParam("system");
+		dc.addParam(new Date());
+		dc.addParam("structure");
+		dc.loadResult();
+
+		//Inserts into Structure Table the Vanity URL Content Type
+		dc = new DotConnect();
+		dc.setSQL(INSERT_STRUCTURE_QUERY);
+		dc.addParam(DEFAULT_VANITY_URL_STRUCTURE_INODE);
+		dc.addParam(DEFAULT_VANITY_URL_STRUCTURE_NAME);
+		dc.addParam(DEFAULT_VANITY_URL_STRUCTURE_DESCRIPTION);
+		dc.addParam(7);
+		dc.addParam(false);
+		dc.addParam(true);
+		dc.addParam(DEFAULT_VANITY_URL_STRUCTURE_VARNAME);
+		dc.addParam("SYSTEM_HOST");
+		dc.addParam("SYSTEM_FOLDER");
+		dc.addParam(new Date());
+		dc.addParam(false);
+		dc.loadResult();
+		
+		//Gets the list of required fields for the VanityURL Content Type
+		List<Map<String,Object>> requiredFields = setBaseVanityURLFields();
+		
+		int i = 0;
+		for(Map<String,Object> field : requiredFields){
+			
+			//Insert into Inode Table the reference of the field
+			dc.setSQL(INSERT_INODE_QUERY);
+			dc.addParam(DEFAULT_VANITY_URL_FIELDS_INODES[i]);
+			dc.addParam("");
+			dc.addParam(new Date());
+			dc.addParam("field");
+			dc.loadResult();
+			
+			//Insert into Field Table
+	        dc.setSQL(INSERT_FIELD_QUERY);
+	        dc.addParam(DEFAULT_VANITY_URL_FIELDS_INODES[i]);
+	        dc.addParam(field.get("structureInode"));
+	        dc.addParam(field.get("fieldName"));
+	        dc.addParam(field.get("fieldType"));
+	        dc.addParam(field.get("fieldRelationType"));
+	        dc.addParam(field.get("fieldContentlet"));
+	        dc.addParam(field.get("isRequired"));
+	        dc.addParam(field.get("isIndexed"));
+	        dc.addParam(field.get("isListed"));
+	        dc.addParam(field.get("velocityVarName"));
+	        dc.addParam(field.get("sortOrder"));
+	        dc.addParam(field.get("values"));
+	        dc.addParam(field.get("regexCheck"));
+	        dc.addParam(field.get("hint"));
+	        dc.addParam(field.get("defaultValue"));
+	        dc.addParam(field.get("isFixed"));
+	        dc.addParam(field.get("isReadOnly"));
+	        dc.addParam(field.get("isSearchable"));
+	        dc.addParam(field.get("isUnique"));
+	        dc.addParam(new Date());
+	        dc.loadResult();
+	        
+	        i++;
+		}
+		
+		
+	}
+
+	/**
+	 * This method creates each field that is required for the Vanity URL Content Type and return a list with them.
+	 * @return list of fields
+	 */
+	private List<Map<String, Object>> setBaseVanityURLFields() {
+		List fields = new ArrayList();
+		
+		fields.add(setFieldElements("Order", "com.dotcms.contenttype.model.field.TextField","", "integer2", true, true, false, "order", 5, "", "", "", "", true, false, true));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "com.dotcms.contenttype.model.field.TextField" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 169 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)